dev/src/write-batch.ts

```diff
@@ -567,25 +614,55 @@ export class WriteBatch {
     );
 
     if (explicitTransaction) {
+      assert(!useBatch, 'Cannot use batch mode with transactions');
       request.transaction = explicitTransaction;
     }
 
-    return this._firestore
-      .request<api.ICommitRequest, api.CommitResponse>('commit', request, tag)
-      .then(resp => {
-        if (request.writes!.length > 0) {
-          const commitTime = Timestamp.fromProto(resp.commitTime!);
-          return resp.writeResults.map(
-            writeResult =>
-              new WriteResult(
-                writeResult.updateTime
-                  ? Timestamp.fromProto(writeResult.updateTime)
-                  : commitTime
-              )
-          );
-        }
-        return [];
-      });
+    if (useBatch) {
+      return this._firestore
+        .request<api.IBatchWriteRequest, api.BatchWriteResponse>(
+          'batchWrite',
+          request as api.IBatchWriteRequest,
```

Comment:
I am leaning towards recommending that you should have two different methods. The new method would be fairly short since we can't support implicit transactions directly.

We should also think about if we need a dummy request for batch writes to make sure the network is initialized.

Reply:
Separated out logic into `batchCommit()` method